framework/src/interfaces/TaggingInterface.C

```diff
@@ -359,8 +374,15 @@ TaggingInterface::prepareMatrixTagLower(Assembly & assembly,
 void
 TaggingInterface::accumulateTaggedLocalResidual()
 {
+  if (_eigen_problem)
+  {
+    Real fac = (_eigen_problem->negativeSignEigenKernel() ? 1 : -1) * (_bx_norm ? *_bx_norm : 1);
```

Comment:
The sign factor should not be applied for Ax kernels, correct?

Reply:
I am a bit abusing `if (_eigen_problem)` here. Only an eigen kernel can get that pointer assigned in the constructor.